src/main/java/core/basesyntax/RobotRoute.java

```diff
@@ -1,7 +1,83 @@
 package core.basesyntax;
 
 public class RobotRoute {
+    private Robot robot;
+    private int toX;
+    private int toY;
+
     public void moveRobot(Robot robot, int toX, int toY) {
-      //write your solution here
+        if (robot == null) {
+            throw new IllegalArgumentException("Robot must be present (not null)");
+        }
+
+        this.robot = robot;
+        this.toX = toX;
+        this.toY = toY;
```

Comment:
What do you think, is it better to just pass this argument to the methods bellow as as an argument instead of saving it to the object fields?

```
this.setXAxisDirection(robot, toX);
this.performXAxisMovement(robot, toX);

this.setYAxisDirection(robot, toY);
this.performYAxisMovement(robot, toY);
```